### ISSUE TITLE
Sampling from exp of a Normal random variable crashes 

### ISSUE THREAD

**Issue by rikardn** [sympy/sympy#20376]
Trying to sample from an exponential of a normally distributed random variable crashes sympy.

To reproduce:

```python
import sympy
import sympy.stats

X = sympy.stats.Normal('X', 0, 0.1)
sympy.stats.sample(sympy.exp(X))  
```

It works if I instead sample using the pspace directly:

```
In [9]: expr.subs(sympy.stats.pspace(expr).sample())                                                                                                                      
Out[9]: 0.991651703984268
```

A session with the error:
```
In [1]: import sympy.stats                                                                                                                                                

In [2]: X = sympy.stats.Normal('X', 0, 0.1)                                                                                                                               

In [3]: sympy.stats.sample(X)                                                                                                                                             
Out[3]: -0.0506707785768209

In [4]: sympy.stats.sample(sympy.exp(X))                                                                                                                                  
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-c82e94cb5490> in <module>
----> 1 sympy.stats.sample(sympy.exp(X))

~/.local/lib/python3.7/site-packages/sympy/stats/rv.py in sample(expr, condition, size, **kwargs)
   1038 
   1039     """
-> 1040     return next(sample_iter(expr, condition, size=size, numsamples=1))
   1041 
   1042 

~/.local/lib/python3.7/site-packages/sympy/stats/rv.py in sample_iter(expr, condition, size, numsamples, **kwargs)
   1077     # lambdify is much faster but not as robust
   1078     try:
-> 1079         return sample_iter_lambdify(expr, condition, size=size, numsamples=numsamples, **kwargs)
   1080     # use subs when lambdify fails
   1081     except TypeError:

~/.local/lib/python3.7/site-packages/sympy/stats/rv.py in sample_iter_lambdify(expr, condition, size, numsamples, **kwargs)
   1152     d = ps.sample(size)  # a dictionary that maps RVs to values
   1153     args = [d[rv] for rv in rvs]
-> 1154     fn(*args)
   1155     if condition:
   1156         given_fn(*args)

<lambdifygenerated-2> in _lambdifygenerated(X)
      1 def _lambdifygenerated(X):
----> 2     return (exp(X))

AttributeError: 'Float' object has no attribute 'exp'
```

**Comment by Maelstrom6** [comment]
Which version are you using? I tried this on current master and version 1.6.1 and couldn't replicate the error.

**Comment by rikardn** [comment]
I am sorry, I was using 1.6.2. I tested the current master and I get a slightly different crash (note the added `next` since sympy now returns a generator):

```python
import sympy
import sympy.stats

X = sympy.stats.Normal('X', 0, 0.1)
next(sympy.stats.sample(sympy.exp(X)))
```

```
  File "<ipython-input-8-87f905e80ab7>", line 1, in <module>
    next(sympy.stats.sample(sympy.exp(X)))

  File "/home/devel/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values

  File "/home/devel/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}

  File "/home/devel/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)

  File "/home/devel/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)

  File "/home/devel/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 954, in rvs
    vals = self._rvs(*args)

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 889, in _rvs
    Y = self._ppf(U, *args)

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 902, in _ppf
    return self._ppfvec(q, *args)

  File "/usr/lib/python3/dist-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)

  File "/usr/lib/python3/dist-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 1587, in _ppf_single
    while self._ppf_to_solve(right, q, *args) < 0.:

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 1569, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 1745, in cdf
    place(output, cond, self._cdf(*goodargs))

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 1621, in _cdf
    return self._cdfvec(x, *args)

  File "/usr/lib/python3/dist-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)

  File "/usr/lib/python3/dist-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)

  File "/home/.local/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py", line 1618, in _cdf_single
    return integrate.quad(self._pdf, self.a, x, args=args)[0]

  File "/home/.local/lib/python3.7/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    points)

  File "/home/.local/lib/python3.7/site-packages/scipy/integrate/quadpack.py", line 450, in _quad
    return _quadpack._qagie(func,bound,infbounds,args,full_output,epsabs,epsrel,limit)

  File "/home/devel/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)

  File "<lambdifygenerated-3>", line 2, in _lambdifygenerated
    return (5.0*sqrt(2)*exp(-50.0*Dummy_23**2)/sqrt(pi))

OverflowError: (34, 'Numerical result out of range')
```

**Comment by Maelstrom6** [comment]
I think this is a SciPy problem. 

Somewhere along the line, SymPy does the following:
```python
from scipy.stats import rv_continuous
import numpy as np

class scipy_pdf(rv_continuous):
    def _pdf(self, x):  # pdf of N(0, 0.1)
        return 5 * np.sqrt(2) * np.exp(-50 * x ** 2) / np.sqrt(np.pi)

scipy_rv = scipy_pdf(a=-np.inf, b=np.inf, name='scipy_pdf')
print(scipy_rv.rvs(size=()))
```

SciPy tries to find the cdf by performing numerical integration from `-np.inf` to `np.inf`. This somehow breaks Python's floating points and it just gives up. The weird thing is that `Normal('X', 0, 1)` works just fine.

I guess this is something that SymPy can "fix" by using finite `a` and `b`. But the hard part will be deciding what it big enough.

As a workaround, I suggest using a more restrictive library like NumPy. You cannot sample with your own handmade distributions if you do this.
```python
from sympy import *
from sympy.stats import *

X = Normal('X', 0, S(1)/10)
print(next(sample(exp(X), library="numpy")))
```
Sampling has been made far more powerful in 1.7, thanks to Smit-create.

**Comment by Smit-create** [comment]
>  I'm not sure how to fix the problem in your first comment for 1.6.2. It has probably been fixed in 1.7 with the improvements to the sampling module.

This is because of https://github.com/sympy/sympy/commit/64d8d439e3bdaa9a4a85aca81aa15f226e188947 on branch 1.7, there is some issue with lambdify and scipy
